File: Coding tests/sample-codes/recursiveBinarySearch.js
function recursiveBinarySearch(target, arr) {
	if (arr.length === 0) {
		return false;
	} else {
		let middle = Math.floor(arr.length / 2);
		if (middle === target) {
			return true;
		} else if (middle < target) {
			return recursiveBinarySearch(target, arr.slice(middle + 1));
		} else {
			return recursiveBinarySearch(target, arr.slice(0, middle));
		}
	}
}

function verify(result) {
	console.log('Target found:', result);
}

const tab = [1, 2, 3, 7, 8, 9, 10];

const res1 = recursiveBinarySearch(7, tab);
const res2 = recursiveBinarySearch(12, tab);

verify(res1);
verify(res2);

File: Coding tests/sample-codes/recursive_binary_search.py
def recursive_binary_search(arr, target):
    if len(arr) == 0:
        return False
    else:
        middle = (len(arr))//2

        if arr[middle] == target:
            return True
        else:
            if arr[middle] < target:
                return recursive_binary_search(arr[middle+1:], target)
            else:
                return recursive_binary_search(arr[:middle], target)


def verify(result):
    print('Target found:', result)


tab = [1, 2, 3, 7, 8, 9, 10]

res1 = recursive_binary_search(tab, 7)
res2 = recursive_binary_search(tab, 14)

verify(res1)
verify(res2)
